lib/Element.js

```diff
@@ -245,13 +246,13 @@ Element.prototype.getBoundingClientRect = function () {
   return this.component.getBoundingClientRect()
 }
 
-Element.prototype.toReact = function (idPrefix, index) {
-  idPrefix = idPrefix || 'faux-dom'
-  index = index || 0
-
+Element.prototype.toReact = function (id) {
   var props = assign({}, this.props)
   props.style = assign({}, props.style)
-  props.key = isUndefined(props.key) ? idPrefix + '-' + index : props.key
+
+  if (isUndefined(props.key) && id) {
```

Comment:
@philpl I think we're also missing a check for `this._key` and need to set it to `props.key` if necessary. Otherwise `this._key` has never been used. Right?